src/walletConnect/screens/useIsDappListed.ts

```diff
@@ -0,0 +1,10 @@
+import { useSelector } from 'react-redux'
+import { dappsListSelector } from 'src/dapps/selectors'
+
+export const useIsDappListed = (dappName?: string) => {
+  const dappsList = useSelector(dappsListSelector)
+
+  return (
+    !!dappName && !!dappsList.find((dapp) => dapp.name.toLowerCase() === dappName.toLowerCase())
+  )
+}
```

Comment:
This could be abused easily. Pretend you're `Ubeswap` and boom 😅 

We should base this on the domain of the dapp url (and callback url for DappKit). 
This may require checking all our current listed dapp urls to be sure they would pass that test.

Reply:
Yes the types are a bit misleading.
The DappKit `callback` is always a url.
For WC, the url is there too, we should use the same logic we use for filling the `dappUrl` for anaytics purpose:
https://github.com/valora-inc/wallet/blob/268b65d121c2038b46dc3778ff35bcd51bb3259d/src/analytics/Properties.tsx#L1029

If for some reason the url is `empty | null | undefined`, we can just deny the listed status.

Reply:
ah cool i didn't know about that! thanks for pointing it out, i've added a check for the url now :D